examples/plot_metrics_with_error.py

```diff
@@ -0,0 +1,188 @@
+z_score = 1.959964
+digits_of_precision = 4
+
+
+def wilson(p, n, digits=digits_of_precision, z=z_score):
+    """ Returns lower and upper bound """
+    denominator = 1 + z**2/n
+    centre_adjusted_probability = p + z*z / (2*n)
+    adjusted_standard_deviation = np.sqrt((p*(1 - p) + z*z / (4*n)))/np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z*adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z*adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+def compute_error_metric(metric_value, sample_size, z_score):
+    """ Standard Error Calculation (Binary Classification)
+
+    Assumes infinitely large population,
+    Should be used when the sampling fraction is small.
+    For sampling fraction > 5%, may want to use finite population correction
+    https://en.wikipedia.org/wiki/Margin_of_error
+
+    Note: 
+        Returns absolute error (%)
+    """
+    return z_score*np.sqrt(metric_value*(1.0-metric_value))/np.sqrt(sample_size)
+
+
+def recall_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    tn, fp, fn, tp = confusion_matrix(y_true, y_pred).ravel()
+    bounds = wilson(tp/(tp+fn), tp + fn, digits_of_precision, z_score)
+    return bounds
+
+def recall_normal_err(y_t, y_p):
+    assert len(y_t) == len(y_p)
+    tn, fp, fn, tp = confusion_matrix(y_t, y_p).ravel()
+    error = compute_error_metric(tp/(tp+fn), tp + fn, z_score=z_score)
+    return (error, error)
+
+def accuracy_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    bounds = wilson(score, len(y_true), digits_of_precision, z_score)
+    return bounds
+
+def accuracy_normal_err(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    error = compute_error_metric(score, len(y_true), z_score)
+    return (error, error)
+
+
+# %%
+# Classifier
+# ----------
+# Now we use a Decision Tree to make predictions
+#
+
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+classifier.fit(X, y_true)
+y_pred = classifier.predict(X)
+
+# %%
+# MetricFrame
+# -----------
+# Now we create a MetricFrame to generate the recall Wilson bounds and recall normal error
+#
+
+# Analyze metrics using MetricFrame
+metrics_dict = {
+    'Recall': recall_score,
+    'Recall Bounds': recall_wilson,
+    'Recall Error': recall_normal_err,
+    'Accuracy': accuracy_score,
+    'Accuracy Bounds': accuracy_wilson,
+    'Accuracy Error': accuracy_normal_err,
+}
+metric_frame = MetricFrame(metrics_dict, y_true, y_pred, sensitive_features=sex)
+
+# %%
+# plot metrics with (symmetric) error bars
+plot_metric_frame(metric_frame, plot_type="scatter", 
+    metrics=['Recall', 'Accuracy'], 
+    error_bars=['Recall Error', 'Accuracy Error'])
+plot_metric_frame(metric_frame, plot_type="bar", metrics='Recall', error_bars='Recall Error')
+
```

Comment:
The first plot looks like this for me: 

<img width="330" alt="image" src="https://user-images.githubusercontent.com/24417440/125804664-16e5da85-c3bd-4579-bd85-a7d60bb671d9.png">

A few remarks
* It doesn't make sense to me that by default this plot includes a color bar; is this an explicit design choice? If not, I'd suggest to remove it by default.
* By default, we might want to items to the legend that indicate what the error bar means (e.g., is it a 95% CI?).
* If we are creating separate confidence intervals for each group and also across multiple metrics, I think we need to adjust for multiple testing. In our case we are particularly interested to see whether there are differences across groups. Even though we are just comparing them visually, I'd argue that's an implicit hypothesis test. This might lead to wrong conclusions about differences between groups that arises due to chance.
* I think we should add an example of how people can scale the y-axis for better comparison across metrics. I probably wouldn't show this figure to other stakeholders, because the y-scale starts at non-zero and the scale is not consistent across metrics (which is not always problematic, but in case of accuracy/recall I'd like to put them on a scale from 0 to 1).

Reply:
The second plot looks like this for me:

<img width="302" alt="image" src="https://user-images.githubusercontent.com/24417440/125806291-17a58b72-587f-42ff-af17-6a9505fa41bb.png">

I believe displaying the numbers was discussed in an earlier call, but I don't recall the conclusion. I don't think I would include them by default, but we could definitely add an example of how to add them.

Reply:
- Hmm when I plot it using VSCode's interactive mode, I don't get the color bars. Are there any configurations you/me might have that cause this? 
-

Reply:
> Hmm when I plot it using VSCode's interactive mode, I don't get the color bars. Are there any configurations you/me might have that cause this?

I generated the plots by building the docs on your fork and inspecting them there (probably not the most efficient way, lol, but they should work there eventually so might be good to check that out earlier rather than later). I'm not sure what exact configurations we're using there. 

> I wrote some code to try to fix this, and it seems the second image (with this code) displays weirder. Can you try to see if this works? Or do you know a better way to do this?

Hmmm... you'd think there should be a way to avoid the color bar from being added in the first place. I'll see if I can replicate the 'issue' in a local IDE.

> Yeah I think including the 95% CI information is good! I can see how that will work for the scatter plots, and for the bar plots I can probably add to the legend a black box for the 95% CI

Awesome!

Reply:
@hildeweerts I got the 95% CI signage in the legend and added an example in the notebook on how to standardize the y-axis. For the multiple testing do we want to assume a certain way to calculate it (since there are multiple ways)? If so, can you send a resource for the formula to calculate it?